clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CatsOnDemandCacheUpdater.groovy

```diff
@@ -93,6 +96,8 @@ class CatsOnDemandCacheUpdater implements OnDemandCacheUpdater {
         log.warn("$agent.providerName/$agent.onDemandAgentType failed to handle on demand update for $type", e)
       }
     }
+
+    return hasOnDemandResults ? OnDemandCacheUpdater.OnDemandCacheStatus.PENDING : OnDemandCacheUpdater.OnDemandCacheStatus.SUCCESSFUL
```

Comment:
There are race conditions at play wherein a force cache refresh could come in at the ~exact same time as a normal caching cycle and be lost.

More specifically, it can take a non-zero amount of time to bundle up and apply the cache writes and that exposes a window of time where an on-demand update could be applied and then be overwritten with stale data.

It's the stale data after a force cache refresh that we need to avoid happening.